package fr.afcepf.al34.authenticator.tester;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import fr.afcepf.al34.authenticator.AuthenticationManager;
import fr.afcepf.al34.authenticator.Credentials;

public class Test {

	private static final Logger logger = LogManager.getLogger();
	
	public static void main(String[] args) throws Exception {
		Credentials c1 = new Credentials("toto");
		Credentials c2 = new Credentials("jane");
		
		AuthenticationManager.initializeCredentials("soleil", c1);
		logger.info(AuthenticationManager.authenticate("solel", c1));
		logger.error("Password not valid !!!");
		AuthenticationManager.authenticate("solel", c1);
		logger.error("Password not valid !!!");
		logger.warn("Trying to connect with incorrect password several times");
		AuthenticationManager.authenticate("soleil", c1);
		logger.info("successful connection");
		
		System.out.println("______________________________");
		
		AuthenticationManager.initializeCredentials("lune", c2);
		AuthenticationManager.authenticate("line", c2);
		logger.info("Password not valid !!!");
		AuthenticationManager.authenticate("lune", c2);
		logger.info("successful connection");
	}
	
}
